### PR TITLE
[DOCU-2047] Consumer groups

### PR DESCRIPTION
**Pull request by lena-larionova**
### Summary
Consumer groups doc.

* Admin API reference for consumer groups
* Admin API examples for consumer groups: cURL and HTTPie examples
* RLA plugin ref: added two new properties

**For reviewers:** ignore the file `app/_hub/kong-inc/rate-limiting-advanced/1.5.x.md` - this is a new file but with no new content, as a side-effect of the plugin doc versioning. Please review `app/_hub/kong-inc/rate-limiting-advanced/index.md` instead.

### Reason
Consumer groups feature.

### Testing

Tested all examples locally with a 2.7 preview build.

[API Reference](https://deploy-preview-3443--kongdocs.netlify.app/gateway/2.7.x/admin-api/consumer-groups/reference/)
[Examples](https://deploy-preview-3443--kongdocs.netlify.app/gateway/2.7.x/admin-api/consumer-groups/examples/)
[RLA plugin param reference](https://deploy-preview-3443--kongdocs.netlify.app/hub/kong-inc/rate-limiting-advanced/)

### CODE REVIEW

File: app/gateway/2.7.x/admin-api/consumer-groups/examples.md [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+2. The plugin you just set up applies to all consumers in the cluster. Let's
+change the settings for the `JL` consumer group only:
+
+{% capture override %}
+{% navtabs codeblock %}
+{% navtab cURL %}
+```bash
+curl -i -X PUT http://{HOSTNAME}:8001/consumer_groups/JL/overrides/plugins/  rate-limiting-advanced \
+  --header 'Content-Type: application/json' \
+  --data '{ "config": { "limit": [ 10 ], "retry_after_jitter_max": 1, "window_size": [ 10 ] } }'
```
Comment (lena-larionova) [inline app/gateway/2.7.x/admin-api/consumer-groups/examples.md:178]: @yzyyi The only way I could get this one to work was with a JSON body. If I used this format (and a bunch of different syntax attempts for the arrays):

```
curl -i -X PUT http://localhost:8001/consumer_groups/JL/overrides/plugins/rate-limiting-advanced \
  -d "config.limit[1]=10" \
  -d "config.window_size[1]=10" \
```

Then I'd always get this error:
```
{"message":"2 schema violations (config.limit: {\n  \"expected a number\"\n}; config.window_size: {\n  \"expected a number\"\n})","name":"schema violation","fields":{"config":{"window_size":["expected a number"],"limit":["expected a number"]}},"code":2}%
```

File: app/gateway/2.7.x/admin-api/consumer-groups/reference.md [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+`GROUP_NAME|GROUP_ID`<br>*required*   | The name or UUID of the consumer group to configure.
+
+**Request Body**
+
+Attribute     | Description  
+---------:    | --------
+`limit`<br>*required*       | An array of one or more requests-per-window limits to apply. There must be a matching number of window limits and sizes specified.
+`window_size`<br>*required* | An array of one or more window sizes to apply a limit to (defined in seconds). There must be a matching number of window limits and sizes specified.
+`window_type`<br>*optional* | Set the time window type to either `sliding` (default) or `fixed`.
+`retry_after_jitter_max`<br>*optional* | The upper bound of a jitter (random delay) in seconds to be added to the `Retry-After` header of denied requests (status = `429`) in order to prevent all the clients from coming back at the same time. The lower bound of the jitter is `0`; in this case, the `Retry-After` header is equal to the `RateLimit-Reset` header.
```
Comment (lena-larionova) [inline app/gateway/2.7.x/admin-api/consumer-groups/reference.md:314]: Are these all of the possible editable params for this endpoint?

Comment (yzyyi) [reply app/gateway/2.7.x/admin-api/consumer-groups/reference.md:314]: Yes.

File: app/gateway/2.7.x/admin-api/consumer-groups/examples.md (hunk 1 above)
Comment (lena-larionova) [reply app/gateway/2.7.x/admin-api/consumer-groups/examples.md]: Resolved in https://github.com/Kong/kong-ee/pull/2810.

File: app/gateway/2.7.x/admin-api/consumer-groups/examples.md [hunk 3]
```diff
@@ -0,0 +1,532 @@
...
+        "custom_id": null,
+        "id": "8089a0e6-1d31-4e00-bf51-5b902899b4cb",
+        "tags": null,
+        "type": 0,
+        "username": "DianaPrince",
+        "username_lower": "dianaprince"
+    }
+    ```
+
+3. Add `DianaPrince` to the `JL` consumer group:
```
Comment (yzyyi) [inline app/gateway/2.7.x/admin-api/consumer-groups/examples.md:100]: Do we want to show an example of adding multiple consumers to a group?
If so, we need to also notify user that we will validate the full list first before assigning:
Consumers need to be valid (existing, and not already in the group)
The assign will error out and error message will contain first element that doesn't comply, no consumer would be assigned in this case.

File: app/gateway/2.7.x/admin-api/consumer-groups/examples.md [hunk 4]
```diff
@@ -0,0 +1,532 @@
...
+                },
+                "created_at": 1638916518,
+                "id": "b7c426a2-6fcc-4bfd-9b7a-b66e8f1ad260",
+                "name": "rate-limiting-advanced"
+            }
+        ]
+    }
+    ```
+
+## Remove consumer from group - group view
```
Comment (yzyyi) [inline app/gateway/2.7.x/admin-api/consumer-groups/examples.md:304]: Shall we show examples of deleting all consumers from a group and delete all groups for a consumer?
I think the API reference page has it but want to double check.

File: app/_hub/kong-inc/rate-limiting-advanced/index.md [hunk 5]
```diff
@@ -285,6 +277,26 @@ params:
...
+    - name: enforce_consumer_groups
+      required: false
+      default: false
+      value_in_examples: true
+      datatype: boolean
+      description: |
+        Set to `true` to enable `consumer_groups`, which allows the settings
+        from one of the allowed consumer groups to override the given plugin
+        configuration.
+    - name: consumer_groups
```
Comment (yzyyi) [inline app/_hub/kong-inc/rate-limiting-advanced/index.md:289]: Can we add a note here (something like)
Flipping `enforce_consumer_groups` from `true` to `false` will disable the group override, but will not clear the list of `consumer_groups` provided here. You can flip `enforce_consumer_groups` to true to re-enforce the groups.

File: app/gateway/2.7.x/admin-api/consumer-groups/examples.md (hunk 3 above)
Comment (lena-larionova) [reply app/gateway/2.7.x/admin-api/consumer-groups/examples.md]: Added a separate example, see https://github.com/Kong/docs.konghq.com/pull/3443/commits/57813ee86dbda51b3ac8485f5bcab327e92c57e4

File: app/gateway/2.7.x/admin-api/consumer-groups/examples.md (hunk 4 above)
Comment (lena-larionova) [reply app/gateway/2.7.x/admin-api/consumer-groups/examples.md:297]: Added.

**Review by falondarville — APPROVED** [review]
LGTM. For Consumer Groups Examples, we should (post-merge) add editable code blocks for the editable text like `{HOSTNAME}`.

**Pull request merged**